webrtc/src/peer_connection/peer_connection_internal.rs

```diff
@@ -924,154 +933,156 @@ impl PeerConnectionInternal {
+        let handled = self.handle_undeclared_ssrc(ssrc, &parsed).await?;
+        if handled {
+            return Ok(());
+        }
 
-                let (mid_extension_id, audio_supported, video_supported) = self
-                    .media_engine
-                    .get_header_extension_id(RTCRtpHeaderExtensionCapability {
-                        uri: ::sdp::extmap::SDES_MID_URI.to_owned(),
-                    })
-                    .await;
-                if !audio_supported && !video_supported {
-                    return Err(Error::ErrPeerConnSimulcastMidRTPExtensionRequired);
-                }
+        // Get MID extension ID
+        let (mid_extension_id, audio_supported, video_supported) = self
+            .media_engine
+            .get_header_extension_id(RTCRtpHeaderExtensionCapability {
+                uri: ::sdp::extmap::SDES_MID_URI.to_owned(),
+            })
+            .await;
+        if !audio_supported && !video_supported {
+            return Err(Error::ErrPeerConnSimulcastMidRTPExtensionRequired);
+        }
 
-                let (sid_extension_id, audio_supported, video_supported) = self
-                    .media_engine
-                    .get_header_extension_id(RTCRtpHeaderExtensionCapability {
-                        uri: ::sdp::extmap::SDES_RTP_STREAM_ID_URI.to_owned(),
-                    })
-                    .await;
-                if !audio_supported && !video_supported {
-                    return Err(Error::ErrPeerConnSimulcastStreamIDRTPExtensionRequired);
-                }
+        // Get RID extension ID
+        let (sid_extension_id, audio_supported, video_supported) = self
+            .media_engine
+            .get_header_extension_id(RTCRtpHeaderExtensionCapability {
+                uri: ::sdp::extmap::SDES_RTP_STREAM_ID_URI.to_owned(),
+            })
+            .await;
+        if !audio_supported && !video_supported {
+            return Err(Error::ErrPeerConnSimulcastStreamIDRTPExtensionRequired);
+        }
 
-                let (rsid_extension_id, _, _) = self
-                    .media_engine
-                    .get_header_extension_id(RTCRtpHeaderExtensionCapability {
-                        uri: SDES_REPAIR_RTP_STREAM_ID_URI.to_owned(),
-                    })
-                    .await;
+        let (rsid_extension_id, _, _) = self
+            .media_engine
+            .get_header_extension_id(RTCRtpHeaderExtensionCapability {
+                uri: SDES_REPAIR_RTP_STREAM_ID_URI.to_owned(),
+            })
+            .await;
+
+        let mut buf = vec![0u8; self.setting_engine.get_receive_mtu()];
+        // Packets that we read as part of simulcast probing that we need to make available
+        // if we do find a track later.
+        let mut buffered_packets: VecDeque<(Bytes, Attributes)> = VecDeque::default();
+
+        let n = rtp_stream.read(&mut buf).await?;
```

Comment:
This read does not go through the interceptor chain hence is missed by all interceptors :( Unsure if we can solve this

Reply:
Idle though: Maybe we can open the SRTP stream earlier in which case we can optimistically configure the interceptor chain...

Reply:
what are the downsides of this read not going through interceptors?

Reply:
We have an interceptor that dumps all ingress RTP, it misses this initial packet. In general any interceptors that reads RTP will miss the first RTP packet

Reply:
It's a bit of a chicken and egg problem however, we only have the necessary information to create the interceptor at line 1001. This information is derived from the payload type which we read out of the first packet.

We'd need a way to fake a read through the interceptor chain to alleviate this I think